Fork in persistent storage implementation to otlp (#4793)
open-telemetry/opentelemetry-dotnet: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/PersistentStorage/README.md

# Persistent Storage APIs for OTLP Exporter

> The files in this folder has been forked in from

Reviewer: ```suggestion
The files in this folder have been copied over from
```

> [OpenTelemetry.PersistentStorage.Abstractions](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/main/src/OpenTelemetry.PersistentStorage.Abstractions)

Reviewer: Use permalink instead of `main` branch:

```suggestion
[OpenTelemetry.PersistentStorage.Abstractions](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/58607b7cdeb15207027a6fa4ca56e7fac897bda4/src/OpenTelemetry.PersistentStorage.Abstractions)
```
